lib/middleware-common/src/call_trace.rs

```diff
@@ -0,0 +1,27 @@
+use wasmer_runtime_core::{
+    codegen::{Event, EventSink, FunctionMiddleware, InternalEvent},
+    module::ModuleInfo,
+};
+
+pub struct CallTrace;
+
+impl FunctionMiddleware for CallTrace {
+    type Error = String;
+    fn feed_event<'a, 'b: 'a>(
+        &mut self,
+        op: Event<'a, 'b>,
+        module_info: &ModuleInfo,
+        sink: &mut EventSink<'a, 'b>,
+    ) -> Result<(), Self::Error> {
+        match op {
+            Event::Internal(InternalEvent::FunctionBegin(id)) => {
+                sink.push(Event::Internal(InternalEvent::Bkpt(Box::new(move |_| {
+                    println!("func ({})", id);
```

Comment:
Could we also add args and print to stderr? Similar to: https://github.com/wasmerio/wasmer/blob/master/lib/clif-backend/src/resolver.rs#L389

Reply:
It is also behind the "debug" feature flag: https://github.com/wasmerio/wasmer/blob/master/lib/clif-backend/src/module_env.rs#L389

Reply:
Changed to print to stderr. It needs some machine-specific tricks to print args and I think we can defer it a bit.